src/vs/workbench/browser/web.api.ts

```diff
@@ -73,6 +73,16 @@ export interface IWorkbench {
 	};
 
 	window: {
+
+		/**
+		 * Logging for embedder.
+		 *
+		 * @param id Human-readable name associated with your embedder.
+		 * @param level The log level of the message to be printed.
+		 * @param message Message to be printed.
+		 */
+		log(id: string, level: LogLevel, message: string): void;
```

Comment:
@tanhakabir What is the purpose of `id`?

Reply:
I wanted to show which embedder extension created the log message. AML vs the Basis flow vs Azure Functions etc...

Reply:
I would wrap that in the message instead of adding `id` in the API. `id` in the API is misleading.